x/gamm/keeper/migrate.go

```diff
@@ -118,28 +121,51 @@ func (k Keeper) validateRecords(ctx sdk.Context, records []types.BalancerToConce
 		}
 
 		// Ensure the provided balancerPoolId exists and that it is of type balancer
-		pool, err := k.GetPool(ctx, record.BalancerPoolId)
+		balancerPool, err := k.GetPool(ctx, record.BalancerPoolId)
 		if err != nil {
 			return err
 		}
-		poolType := pool.GetType()
+		poolType := balancerPool.GetType()
 		if poolType.String() != "Balancer" {
 			return fmt.Errorf("Balancer pool ID #%d is not of type balancer", record.BalancerPoolId)
 		}
 
 		// Ensure the provided ClPoolId exists and that it is of type concentrated.
 		// If clPoolID is 0, this signals a removal, so we skip this check.
+		var clPool poolmanagertypes.PoolI
 		if record.ClPoolId != 0 {
-			pool, err = k.clKeeper.GetPool(ctx, record.ClPoolId)
+			clPool, err = k.clKeeper.GetPool(ctx, record.ClPoolId)
 			if err != nil {
 				return err
 			}
-			poolType = pool.GetType()
+			poolType = clPool.GetType()
 			if poolType.String() != "Concentrated" {
```

Comment:
Presumably we have an actual enum for this somewhere, right? Can we do that instead of string matching?

Reply:
We do have an enum, but externally its less clear. This is quite clear that we want the pool to be of type Concentrated instead of type 3, which could mean anything.

Reply:
Addressed this here, where we now compare it to type instead of string

https://github.com/osmosis-labs/osmosis/pull/4126/commits/1844dcc1c66708cad48590e9cae9cbbdb1c25202

CC @p0mvn addressing your comment